NYPL/scsbuster
# TGR-105  update Ruby version + GHA

## JIRA tickets:
- [SCSBuster - update Ruby version + GHA](https://newyorkpubliclibrary.atlassian.net/browse/TGR-105)

## Things Done:
- Updated OS / Ruby / Rails / Gems
- Implemented GitHub Actions for PR builds
- Implemented GitHub Actions for deployments
- Added RAILS_ENV=production to the deployed instances
- Set `qa` as the default branch and added branch protection rules to `qa` and `production`

## How to Test:
- Run the unit tests
- Build the app locally via README instructions.
- Log into the app at localhost:3001. Make sure it redirects as expected. If your email address is in the super secret list, you'll be allowed in.
- The image has already been deployed to https://scsbuster-qa.nypl.org as evidence that the deploy pipeline works, so you can check it out there as well.
